src/Sylius/Bundle/ResourceBundle/Controller/ParametersParser.php

```diff
@@ -77,6 +78,16 @@ private function parseRequestValue($parameter, Request $request)
             return $this->parseRequestValueExpression(substr($parameter, 5), $request);
         }
 
+        if (0 === strpos($parameter, '!!')) {
+            $parameter = explode(' ', $parameter);
+
+            $parser = trim($parameter[0], '!').'val';
+
+            Assert::oneOf($parser, ['intval', 'floatval', 'doubleval'], 'Variable can be parsed only to int, float or double.');
+
+            return $parser($request->get(substr($parameter[1], 1)));
```

Comment:
Anyway, it will fail if you try to typecast an expression like `!!float expr:abc`, can we try something like the following one instead:

```
return $parser($this->parseRequestValue($parameter[1], $request));
```